hera_cal/datacontainer.py

```diff
@@ -517,19 +518,30 @@ class RedDataContainer(DataContainer):
     '''Structure for containing redundant visibilities that can be accessed by any
         one of the redundant baseline keys (or their conjugate).'''
 
-    def __init__(self, data, reds=None, antpos=None, bl_error_tol=1.0):
+    def __init__(self, data, reds: Sequence[Sequence[Baseline | AntPair]]=None, antpos: dict[int, np.ndarray] | None=None, bl_error_tol: float=1.0):
         '''Creates a RedDataContainer.
 
-        Arguments:
-            data: DataContainer or dictionary of visibilities, just as one would pass into DataContainer().
-                Will error if multiple baselines are part of the same redundant group.
-            reds: list of lists of redundant baseline tuples, e.g. (ind1, ind2, pol).
-            antpos: dictionary of antenna positions in the form {ant_index: np.array([x, y, z])}.
-                Will error if one tries to provide both reds and antpos. If neither is provided,
-                will try to to use data.antpos (which it might have if its is a DataContainer).
-            bl_error_tol: the largest allowable difference between baselines in a redundant group
-                (in the same units as antpos). Normally, this is up to 4x the largest antenna position
-                error. Will only be used if reds is inferred from antpos.
+        Parameters
+        ----------
+        data: DataContainer or dictionary of visibilities, just as one would pass into DataContainer().
+            Will error if multiple baselines are part of the same redundant group.
+        reds: list of lists of redundant baseline tuples, e.g. (ind1, ind2, pol).
+        antpos: dictionary of antenna positions in the form {ant_index: np.array([x, y, z])}.
+            Will error if one tries to provide both reds and antpos. If neither is provided,
+            will try to to use data.antpos (which it might have if its is a DataContainer).
+        bl_error_tol : float
+            the largest allowable difference between baselines in a redundant group
+            (in the same units as antpos). Normally, this is up to 4x the largest antenna position
+            error. Will only be used if reds is inferred from antpos.
+
+        Attributes
+        ----------
+        reds 
+            list of lists of redundant baseline tuples, e.g. (ind1, ind2, pol).
+        red_groups
+            A :class:`RedundantGroups` object that contains the redundant groups for 
+            the entire array, and methods to manipulate them. 
+            When possible, you should use this object instead of the reds attribute.
```

Comment:
Having both is very confusing. I thought the idea was for the new object to be a drop-in replacement.

Reply:
Yeah, I originally replaced `reds` with the `RedundantGroups` object, but came across a few errors. I think with my additional understanding of how it works now (and the addition of the `.append()` method on the class) I could exactly replace it.